feature/experimental/p4rt/otg_tests/traceroute_packetout_test/traceroute_packetout_test.go

```diff
@@ -264,22 +244,100 @@ func TestPacketOut(t *testing.T) {
 		t.Fatalf("Couldn't parse router MAC: %v", err)
 	}
 
-	args := &testArgs{
-		ctx:    ctx,
-		leader: leader,
-		dut:    dut,
-		ate:    ate,
-		top:    top,
-		srcMAC: srcMAC,
-		dstMAC: dstMAC,
+	if err := setupP4RTClient(ctx, client); err != nil {
+		t.Fatalf("Could not setup p4rt client: %v", err)
 	}
 
-	if err := setupP4RTClient(ctx, args); err != nil {
-		t.Fatalf("Could not setup p4rt client: %v", err)
+	for _, m := range genMetadataCombinations() {
+		for _, ipv6 := range []bool{false, true} {
+			args := &testArgs{
+				ctx:         ctx,
+				client:      client,
+				dut:         dut,
+				ate:         ate,
+				top:         top,
+				srcMAC:      srcMAC,
+				dstMAC:      dstMAC,
+				metadata:    m,
+				useIpv6:     ipv6,
+				trafficPort: getExpectedTrafficPort(m),
+				packetIO:    getTracerouteParameter(t),
+			}
+
+			t.Run(args.testName(), func(t *testing.T) {
+				args.packetIO = getTracerouteParameter(t)
+				testPacketOut(ctx, t, args)
+			})
+		}
 	}
+}
+
+// generates all possible combinations of metadata
+// including omission
+func genMetadataCombinations() [][]*p4v1.PacketMetadata {
+	combinations := [][]*p4v1.PacketMetadata{{}} // no metadata
+
+	egressOptions := []string{fmt.Sprint(egressPortId), "TBD BY SWITCH"}
+	submitToIngressOpts := []byte{0, 1}
 
-	args.packetIO = getTracerouteParameter(t)
-	testPacketOut(ctx, t, args)
+	// singletons
+	for _, pId := range egressOptions {
+		combinations = append(combinations, []*p4v1.PacketMetadata{{
+			MetadataId: uint32(1), // "egress_port"
+			Value:      []byte(pId),
+		}})
+	}
+
+	for _, submitIngress := range submitToIngressOpts {
+		combinations = append(combinations, []*p4v1.PacketMetadata{{
+			MetadataId: uint32(2), // "submit_to_ingress"
+			Value:      []byte{submitIngress},
+		}})
+	}
+
+	// binary combinations
+	for _, pId := range egressOptions {
+		for _, submitIngress := range submitToIngressOpts {
+			combinations = append(combinations, []*p4v1.PacketMetadata{
+				{
+					MetadataId: uint32(1), // "egress_port"
+					Value:      []byte(pId),
+				},
+				{
+					MetadataId: uint32(2), // "submit_to_ingress"
+					Value:      []byte{submitIngress},
+				},
+			})
+		}
+	}
+
+	// add ternary combinations with padding
+	for i := len(combinations) - 1; i >= 0; i-- {
+		combinations = append(combinations, append([]*p4v1.PacketMetadata{{
+			MetadataId: uint32(3), // "unused padding"
+			Value:      []byte("0"),
```

Comment:
this should be []byte(0) and []byte(1) is the other value

Reply:
Updated to []byte{0}. What other value are you talking about? this is just the padding. The other option is that this is not set at all.